include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -226,19 +226,46 @@ namespace internal
 
     // helper function
     const auto sort_constraints = [&]() {
+      if (locally_relevant_constraints.empty())
+        return;
+
       std::sort(locally_relevant_constraints.begin(),
                 locally_relevant_constraints.end(),
                 [](const auto &a, const auto &b) {
                   return std::get<0>(a) < std::get<0>(b);
                 });
 
-      locally_relevant_constraints.erase(
-        std::unique(locally_relevant_constraints.begin(),
-                    locally_relevant_constraints.end(),
-                    [](const auto &a, const auto &b) {
-                      return std::get<0>(a) == std::get<0>(b);
-                    }),
-        locally_relevant_constraints.end());
+      auto read_ptr  = locally_relevant_constraints.begin();
+      auto write_ptr = locally_relevant_constraints.begin();
+      while (++read_ptr != locally_relevant_constraints.end())
+        {
+          if (std::get<0>(*read_ptr) != std::get<0>(*write_ptr))
+            {
+              if (++write_ptr != read_ptr)
+                *write_ptr = std::move(*read_ptr);
+            }
```

Comment:
I think we need to describe what happens here: We try to remove duplicates by going through the `locally_relevant_constraints` field after sorting it. If we see that the global `dof_index` as the first entry in the tuple is different from the read position, we pick the entry at the read position.